edgelet/edgelet-core/src/settings.rs

```diff
@@ -292,22 +317,98 @@ impl Listen {
 
 #[derive(Clone, Debug, serde_derive::Deserialize, serde_derive::Serialize)]
 pub struct Certificates {
-    device_ca_cert: PathBuf,
-    device_ca_pk: PathBuf,
-    trusted_ca_certs: PathBuf,
+    device_ca_cert: String,
+    device_ca_pk: String,
+    trusted_ca_certs: String,
+}
+
+fn is_supported_uri(uri: &Url) -> bool {
+    if uri.scheme() == "file" && uri.port().is_none() && uri.query().is_none() {
+        if let Some(host) = uri.host_str() {
+            return "localhost" == host;
+        }
+        return true;
+    }
+    false
+}
+
+fn get_path_from_uri(uri: &Url, variable: &'static str) -> Result<PathBuf, Error> {
+    if is_supported_uri(&uri) {
+        let path = uri
+            .to_file_path()
+            .map_err(|_| ErrorKind::InvalidSettingsUriFilePath(uri.to_string(), variable))?;
```

Comment:
Since the Error returned is a unit, with_context is not usable here. Your point about capturing the error is well taken and I updated it to handling the unit () as opposed to _.